Fix global keystone routing, and cache issues

notque opened the pull request.

olandr commented on pkg/keystone/keystone_test.go:
```diff
@@ -4,11 +4,17 @@
 package keystone
```
Like using some kind of iterator logic:

```
for _, instanceType := [regional, global] {
// testing logic here
}
```

olandr replied:
Hmm, I just noticed the cross-check of the cache keys. I guess that makes it a bit more annoying to reduce this into a simple for-loop. But maybe it could still be an interesting thing to look into?

notque replied:
Yeah, you're absolutely on the right track. This is quite a hack. And so it's very non obvious. 

The problem I have here is the request of what needs to occur, vs what makes sense. I wish I had something more reasonable to say here, but I'm just as flummoxed by the situation.

notque commented on pkg/api/v1api.go:
```diff
@@ -37,27 +36,8 @@ func NewV1Handler(keystoneDriver keystone.Driver, storageDriver storage.Driver)
-			next.ServeHTTP(w, req.WithContext(ctx))
-		})
-	})
+	// Note: Keystone resolution is handled by keystoneResolutionMiddleware at router level
+	// This eliminates race conditions by ensuring consistent keystone selection throughout request lifecycle
```
I considered this. it's a good question. This is where it was before. It was on my mind to potentially remove it. Very valid feedback. I recognized the issue, but didn't make a decision.

notque commented on pkg/api/server.go:
```diff
@@ -108,6 +113,8 @@ func setupRouter(keystoneDriver, globalKeystoneDriver keystone.Driver, storageDr
 
 var validDomain = regexp.MustCompile(`^[a-zA-Z0-9_-]+$`)
 
+const trueValue = "true"
```
I think that's fair feedback. The problem was that this came from our linting rules. So the alternative is to add nolint to the lines where it occurs. I considered this, and decided that the more nolint rules I have is worse than the silliness of having a const for four trivial spots.

notque commented on pkg/api/util.go:
```diff
@@ -369,23 +378,100 @@ func authorize(wrappedHandlerFunc func(w http.ResponseWriter, req *http.Request)
+		// Determine keystone type early and consistently
+		keystoneType, keystoneDriver := determineKeystoneForRequest(r)
+
+		// Set both type and instance in request context
+		ctx := context.WithValue(r.Context(), keystoneTypeKey, keystoneType)
+		ctx = context.WithValue(ctx, keystoneInstanceKey, keystoneDriver)
```
Fair, appreciate it.

notque commented:
@olandr i've addressed your review, please push back on me if you still have concerns. Happy to adjust, but my logic as to why has been added and documented.

olandr reviewed (approved):
> @olandr i've addressed your review, please push back on me if you still have concerns. Happy to adjust, but my logic as to why has been added and documented.

Your responses and small edits are valid in my opinion.

In regard to the keystone_test.go I believe that it would be good to reduce the file at some point. But for the time being this is fine by me I would say.
